daemon/list.go

```diff
@@ -131,12 +149,27 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 		if !psFilters.Match("status", container.State.StateString()) {
 			return nil
 		}
+
+		if ancestorFilter {
+			shouldSkip := true
+			for _, imageIDtoMatch := range imagesFilter {
```

Comment:
If 'imagesFilter' where a map then I think you might be able to replace this for-loop with a hash lookup.